test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -2046,4 +2046,124 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
       });
     });
   });
+
+  context('16. Rewrap', function () {
+    const masterKeys = {
+      aws: {
+        region: 'us-east-1',
+        key: 'arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0'
+      },
+      azure: {
+        keyVaultEndpoint: 'key-vault-csfle.vault.azure.net',
+        keyName: 'key-name-csfle'
+      },
+      gcp: {
+        projectId: 'devprod-drivers',
+        location: 'global',
+        keyRing: 'key-ring-csfle',
+        keyName: 'key-name-csfle'
+      },
+      kmip: {},
+      local: undefined
+    };
+    let client1, client2;
+
+    /**
+     * Run the following test case for each pair of KMS providers (referred to as ``srcProvider`` and ``dstProvider``).
+     * Include pairs where ``srcProvider`` equals ``dstProvider``.
+     */
+    function* generateTestCombinations() {
+      const providers = Object.keys(masterKeys);
+      for (const srcProvider of providers) {
+        for (const dstProvider of providers) {
+          yield { srcProvider, dstProvider };
+        }
+      }
+    }
+
+    beforeEach(async function () {
+      client1 = await this.configuration.newClient().connect();
+      client2 = await this.configuration.newClient().connect();
+    });
```

Comment:
No need to connect anymore woohoo!

Reply:
If this is a blocking comment I'll fix it but otherwise I'll leave it as-is

Reply:
Yea please remove it TIA

Reply:
mongodb-client-encryption doesn't pin a version of the driver - I can't think of why this may be a problem but do you see any concern with testing a feature only supported in 4.7.0+ here?

Reply:
Do we ever test mongodb-client-encryption with versions of the driver other than latest actually?  I don't think so.  

I'm not sure which features of the driver we use in libmongocrypt when using the client but if it's anything non-standard (not stable API) could it be valuable to add multiple driver versions to our libmongocrypt CI?